lib/bolt/cli.rb

```diff
@@ -654,8 +654,21 @@ def generate_types
     # Initializes a specified directory as a Bolt project and installs any modules
     # specified by the user, along with their dependencies
     def initialize_project
-      project    = Pathname.new(File.expand_path(options[:object] || Dir.pwd))
-      config     = project + 'bolt.yaml'
+      name       = options[:object] || File.basename(Dir.pwd)
+      name_regex = /^[a-z][a-z0-9_]*$/
+      if name !~ name_regex
+        if options[:object]
+          raise Bolt::ValidationError, "The provided project name '#{name}' is invalid; "\
+            "project name must match #{name_regex.inspect}"
+        else
+          raise Bolt::ValidationError, "The current directory name '#{name}' will be used "\
+            "as the project name, and is invalid. Project name must match #{name_regex.inspect}"
+        end
+      end
+
+      project    = Pathname.new(Dir.pwd)
+      old_config = project + 'bolt.yaml'
+      config     = project + 'bolt-project.yaml'
```

Comment:
Since we use `'bolt.yaml'` and `'bolt-project.yaml'` in several locations, I wonder if it makes sense to define those as a constant in `Bolt::Project`, similar to `Bolt::Project::BOLTDIR_NAME`: https://github.com/puppetlabs/bolt/blob/main/lib/bolt/project.rb#L9

Reply:
If we do I don't think it should be here - we use those a lot throughout the codebase. On the minus side, `Bolt::Project::BOLT_CONFIG` is way more verbose than `bolt.yaml`.

Reply:
I can't think of names to give those constants that wouldn't be confusing. `'bolt.yaml'` and `'bolt_project.yaml'` are at least obvious when you see them written.